src/Badge.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+
+import { Box, IBox, useTheme } from './';
+
+export enum BadgeVariant {
```

Comment:
Don't think we need to go the other direction, so can prob use `const enum` both here and in `BadgeColor`.

Reply:
No no why wouldn't you export these?

Reply:
Oh yes did not mean to say no export - `export const enum`

Reply:
First of all I get this when I change it to const enum:
`TS2475: 'const' enums can only be used in property or index access expressions or the right hand side of an import declaration or export assignment or type query.`

From what I read const enum makes enum values inlined during compilation. Because of that the code that uses this library won't work - it will compile with BadgeVariant but during runtime it will be undefined.

So finally when I use `const enum` it will work inside my library, but won't work outside.

Here is an [example](https://www.typescriptlang.org/play/index.html#src=enum%20EdgeType%20%7B%0D%0A%20%20%20%20RefersTo%20%3D%200%2C%0D%0A%20%20%20%20Generic%20%3D%2099%0D%0A%7D%20%0D%0A%0D%0Aconst%20enum%20ConstEdgeType%20%7B%0D%0A%20%20%20%20RefersTo%20%3D%200%2C%0D%0A%20%20%20%20Generic%20%3D%2099%0D%0A%7D%0D%0A%0D%0Aconsole.log(EdgeType.Generic)%3B%0D%0Aconsole.log(ConstEdgeType.Generic)%3B)

@XVincentX ^ that may interest you as well.

Reply:
If we want the cake and eat it too, we should be exporting a simple type:

```ts
type BadgeVariant = 'pill' | 'textual'
```

Reply:
Not sure if there is a lot to fight for.

- I think `enum` is handy because we hide things under nice labels :)
- we can also iterate over them (e.g. in storybook) :)
- type is not handy and it's string :(
- `const enum` inlines so cannot be used in libs :(

Reply:
Ok, let's leave as is then!